megameklab/src/megameklab/util/UnitPrintManager.java

```diff
@@ -155,23 +155,23 @@ public static List<PrintRecordSheet> createSheets(List<? extends BTObject> entit
                     }
                 } else if (unit instanceof BattleArmor) {
                     baList.add((BattleArmor) unit);
-                    if (singlePrint || baList.size() > 4) {
+                    if (singlePrint || PrintSmallUnitSheet.fillsSheet(baList, options)) {
                         PrintRecordSheet prs = new PrintSmallUnitSheet(baList, pageCount, options);
                         pageCount += prs.getPageCount();
                         sheets.add(prs);
                         baList = new ArrayList<>();
                     }
                 } else if (unit instanceof Infantry) {
                     infList.add((Infantry) unit);
-                    if (singlePrint || infList.size() > (options.showReferenceCharts() ? 2 : 3)) {
+                    if (singlePrint || PrintSmallUnitSheet.fillsSheet(infList, options)) {
```

Comment:
And here. Though you probably get the point.

Reply:
Same here. `a || b` is fine, it doesn't need to be `(a) || (b)` unless `a` and `b` themselves have some operator in them.